src/runtime/components/sanity-content.ts

```diff
@@ -107,9 +108,20 @@ function findSerializer (item: Block | undefined, serializers: Required<Serializ
   return item?._type ? serializers.types[item._type] || serializers.marks[item._type] : undefined
 }
 
-function renderStyle ({ style, listItem }: Block, serializers: Required<Serializers>, children?: () => Children) {
-  if (!listItem && style && serializers.styles[style]) {
-    return h(serializers.styles[style] as any, null, isVue2 ? children?.() : { default: children })
+function renderStyle (item: Block, serializers: Required<Serializers>, children?: () => Children) {
+  const { style, listItem } = item
+  const serializer = serializers.styles[style]
+  const isElement = typeof serializer === 'string'
+  const props = Object.fromEntries(Object.entries(item).filter(([key]) => key !== '_type' && key !== 'markDefs' && key !== 'children').map(([key, value]) => {
+    if (key === '_key')
+      return ['key', value || null]
+    if (!isElement || validAttrs.includes(key))
+      return [key, value]
+    return []
+  }))
+
+  if (!listItem && style && serializer) {
+    return h(serializer as any, props, isVue2 ? children?.() : { default: children })
   }
```

Comment:
```suggestion
function renderStyle (item: Block, serializers: Required<Serializers>, children?: () => Children) {
  const serializer = item.style && serializers.styles[item.style]
  const isElement = typeof serializer === 'string'
  const props = Object.fromEntries(
    Object.entries(item)
      .filter(([key]) => key !== '_type' && key !== 'markDefs' && key !== 'children')
      .map(([key, value]) => {
        if (key === '_key') return ['key', value || null]
        if (!isElement || validAttrs.includes(key)) return [key, value]
        return []
      }),
  )

  if (!item.listItem && item.style && serializer) {
    return h(serializer as any, props, isVue2 ? children?.() : { default: children })
  }
```

Reply:
next time, please do tick the box to allow maintainers to commit changes directly to the PR branch. I'm remaking this PR ~> https://github.com/nuxt-modules/sanity/pull/523

Reply:
Sorry, didn't realise that was a thing, will do in the future! :)

Reply:
No problem at all. Also very happy for you to pull in my commits on the other branch into this one so you get proper credit ❤️

Reply:
I think I've managed to do that! 😅